streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -648,81 +587,118 @@ public static StreamThread create(final InternalTopologyBuilder builder,
-        final TaskManager taskManager = new TaskManager(changelogReader,
-                                                        logPrefix,
-                                                        restoreConsumer,
-                                                        activeTaskCreator,
-                                                        standbyTaskCreator,
-                                                        new AssignedStreamsTasks(logContext),
-                                                        new AssignedStandbyTasks(logContext));
-
-        return new StreamThread(builder,
-                                clientId,
-                                threadClientId,
-                                config,
-                                processId,
-                                time,
-                                streamsMetadataState,
-                                taskManager,
-                                streamsMetrics,
-                                clientSupplier,
-                                restoreConsumer,
-                                stateDirectory);
+        StreamsMetricsThreadImpl streamsMetrics = new StreamsMetricsThreadImpl(metrics,
+                "stream-metrics",
+                "thread." + threadClientId,
+                Collections.singletonMap("client-id", threadClientId));
+
+        final ThreadCache cache = new ThreadCache(logContext, cacheSizeBytes, streamsMetrics);
 
+        final AbstractTaskCreator<StreamTask> activeTaskCreator = new TaskCreator(builder,
+                config,
+                streamsMetrics,
+                stateDirectory,
+                streamsMetrics.taskCreatedSensor,
+                changelogReader,
+                cache,
+                time,
+                clientSupplier,
+                threadProducer,
+                threadClientId,
+                log);
+        final AbstractTaskCreator<StandbyTask> standbyTaskCreator = new StandbyTaskCreator(builder,
+                config,
+                streamsMetrics,
+                stateDirectory,
+                streamsMetrics.taskCreatedSensor,
+                changelogReader,
+                time,
+                log);
+        TaskManager taskManager = new TaskManager(changelogReader,
+                processId,
+                logPrefix,
+                restoreConsumer,
+                streamsMetadataState,
+                activeTaskCreator,
+                standbyTaskCreator,
+                new AssignedStreamsTasks(logContext),
+                new AssignedStandbyTasks(logContext));
 
+        log.info("Creating consumer client");
+        final String applicationId = config.getString(StreamsConfig.APPLICATION_ID_CONFIG);
+        final Map<String, Object> consumerConfigs = config.getConsumerConfigs(applicationId, threadClientId);
+        consumerConfigs.put(StreamsConfig.InternalConfig.TASK_MANAGER_FOR_PARTITION_ASSIGNOR, taskManager);
```

Comment:
Can't we hand the `taskManager` to `StreamThread` that hands it to `StreamPartitionAssignor`, similar to `adminClient` and remove the internal config?